### PR TITLE
add YAML::Any method to_json

### PR DESCRIPTION
**Pull request by icyleaf**
Add method .to_json for YAML::Any

```crystal
doc = YAML.parse(%(---
        string: hello crystal
        integer: 10
        float: 1.0
        true: true
        false: false
        array:
          - item1
          - item2
        hash:
          key1: value1
          key2: value2
        empty_value:
      ))

puts doc.to_json
```

output: 

```json
{"string":"hello crystal","integer":10,"float":1.0,"true":true,"false":false,"array":["item1","item2"],"hash":{"key1":"value1","key2":"value2"},"empty_value":null}
```

Is this can be add to std method, and how about to_yaml for JSON::Any add too?

### CODE REVIEW

File: src/json/to_json.cr [hunk 1]
```diff
@@ -128,6 +128,37 @@ struct Time
...
+    when "true"
+      json.bool(true)
+    when "false"
+      json.bool(false)
+    when ""
+      json.null
+    when Nil
+      json.null
+    when /^[+-]?([0-9]*[.])?[0-9]+$/
+      object.includes?(".") ? json.number(object.to_f) : json.number(object.to_i)
```
Comment (jhass) [inline src/json/to_json.cr]: This place doesn't feel quite right for this kind of conversions (the `when "true"` and `when "false"`) too, shouldn't `YAML.parse` do them?

Comment (icyleaf) [inline src/json/to_json.cr]: YAML only accepts string, array and hash, see details in parser of yaml and http://www.yaml.org/spec/1.2/spec.html#id2805071

Comment (jhass) [inline src/json/to_json.cr]: Then maybe we shouldn't do this at all, YAML can only emit strings so any YAML being converted JSON can only return JSON of the types that are valid in YAML. The conversion of JSON -> YAML -> JSON simply is lossy by the fact that type wise YAML is a subset of JSON.

Comment (icyleaf) [inline src/json/to_json.cr]: How about rewrite a basic object for json and yaml, because many same methods both in json and yaml ?

**Comment by RX14** [comment]
Why was this closed?

**Comment by icyleaf** [comment]
agree with @jhass, i will try to think how about define a super class to json and yaml which it may be not a struct.

**Pull request closed without merging**